Bundler/Processor/ProcessorService/AssetCreator/FileProcessor.cs

```diff
@@ -0,0 +1,280 @@
+                        assetInfo.Id = assetId;
+                    assetInfo.Version += 1;
+                    assetInfo.OwnerId = userFolderName;
+                    assetInfo.Updated = DateTime.Now.ToString();
+                    UpdateAssetInfoStatus("waiting for update");
+                }
+                else
+                {
+                    // New asset.
+                    assetInfo.Id = assetId;
+                    assetInfo.OwnerId = userFolderName;
+                    assetInfo.Name = zipFileName;
+                    assetInfo.Created = DateTime.Now.ToString();
+                    assetInfo.Updated = assetInfo.Created;
+                    assetInfo.Thumbnail = zipFileName + ".png";
+                    assetInfo.Bundles = new List<AssetCatalog.LODBundle>();
+                    UpdateAssetInfoStatus("waiting");
+                }
+            }
+        }
+
+        private void SetupFolders()
+        {
+            userFolderName = new DirectoryInfo(new FileInfo(zipFileToProcess).DirectoryName).Parent.Name;
+            baseFolderName = new DirectoryInfo(new FileInfo(zipFileToProcess).DirectoryName).Parent.Parent.Name;
+
+            baseFolderPath = Settings.Default.BaseDataPath + "\\" + baseFolderName;
+            processedOutputFolderPath = baseFolderPath + "\\" +  userFolderName + "\\Processed\\"  + assetId;
+            failedOutputFolderPath = baseFolderPath + "\\" + userFolderName + "\\Failed\\" + assetId;
+            tempFolderPath = Settings.Default.TempPath + "\\" + assetId;
+
+            if (Settings.Default.CreateUnityAssetBundles)
+            {
+                bundleOutputFolderPath = baseFolderPath + "\\" + userFolderName + "\\Bundles\\" +  assetId;
+                Directory.CreateDirectory(bundleOutputFolderPath);
+            }
+        }
+
+        private void UpdateAssetInfoStatus(string newStatus)
+        {
+            if (!Settings.Default.CreateUnityAssetBundles)
+                return;
+
+            assetInfo.Status = newStatus;
+            AssetCatalog.UpdateAssetInfo(assetInfo, bundleOutputFolderPath + "\\bundleInfo.json");
+        }
+
+        public async Task<bool> ProcessZip()
+        {
+            busy = true;
+            return await Task.Run(async () =>
+            {
+                if (!File.Exists(zipFileToProcess))
+                {
+                    Console.ForegroundColor = ConsoleColor.Red;
+                    Console.WriteLine(" *** File missing!");
+                    Console.ResetColor();
+                    busy = false;
+                    return false;
+                }
+
+                UpdateAssetInfoStatus("processing_locked");
+
+                var success = false;
+
+                var fileInfo = new FileInfo(zipFileToProcess);
+                while (Utilities.IsFileLocked(fileInfo)) { Task.Delay(100).Wait(); Console.WriteLine("Waiting... {0}", zipFileToProcess); }
+
+                var exportPath = tempFolderPath + "\\Export\\";
```

Comment:
Consider using C# string literals to make reading easier (ex: @"\Export\"). also, i would recommend using Path.Combine here